### DISCUSSION TITLE
Remove some or all Travis CI jobs

### DISCUSSION THREAD

**Discussion by PhysSong** [LMMS/lmms#5802] [Project Management]
Recently, Travis CI announced [their new pricing model](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing). Due to some abusers, they are making some changes which affects our CI jobs.
Some devs talked on Discord to work around this change. 

| job | status |
| --- | --- |
| ShellCheck | Covered by CircleCI but with silghtly different script |
| Linux x86_64, GCC | Covered by CircleCI |
| Debian x86_64, GCC | Travis only |
| Debian i386, GCC | Travis only |
| Debian i386, Clang | Travis only |
| macOS x86_64, AppleClang | Covered by CircleCI |

Things to discuss:
- Do we need to keep jobs covered by CircleCI?
- Is it worth running Debian builds on our side? Can we remove some of those jobs? (CC @jasp00)

<details>
<summary>Relevant discussion on Discord</summary>

> liushuyu said:
> macOS build will be 50 credits per minute from now on
> https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
> OSS projects will get 10k credits at the beginning (not replenish-able) and when runs out, you need to contact support to get more
> At this rate, our credits will run out in ~4 builds
> (Unless someone want to negotiate with the support or warm-up their credit card)
>
> tresf said:
> > Unless someone want to negotiate with the support or warm-up their credit card
>
> Most of our stuff is over on Circle-CI now, but if the pricing isn't too bad, we can consider it.
>
> liushuyu said:
> USD$15 for 25k credits, macOS build costs 50 credits per minute. Timeout once = 2.5k credits spent
> That's $15 for 10 commits
> I don't think this is a good deal at all
> 
> PhysSong said:
> I'd say let's remove the Travis macOS builds.
>
> Dom said:
> My understanding of that Travis blog post is that the OSS credit system affects all build platforms, not just macOS. Even without macOS builds, we still use about an hour and a half of build time per commit. With Linux builds at 10 credits/minute, that's still only 11 commits before we have to go ask for more.
> 
> liushuyu said:
> Correct
> Also, I am curious why we are still having Travis CI running when we already got Circle CI?
>
> PhysSong said:
> No ideas. There are some jobs only running on Travis, but I think we can move to Circle if desired.
> Debian SID .deb builds are not in Circle, but Why are we keeping them?
> i386, Clang builds might be worth keeping, though.
> @jasp00 has added them, so might be worth asking him on a GitHub issue first.
</details>

Summary of the discussion:
- Keep using Travis will require contacting support whenever we run out of credits.
- macOS builds will consume a lot of credits, but the consumption is not small even without it.
- We may consider removing some jobs already covered by CircleCI.

**Comment by jasp00** [comment]
> Is it worth running Debian builds on our side?

One reason for these builds was to help Debian maintenance, which was valid when I had some weight in this maintenance. So you should ask @sebastinas about this issue.

**Comment by sebastinas** [comment]
As long as one Linux build is tested, this should be fine. Paging @snd1 who is probably more interested in LMMS than I am.

**Comment by PhysSong** [comment]
We don't have many days left:
> Please be aware travis-ci.org will be shutting down in several weeks, with all accounts migrating to travis-ci.com. Please stay tuned here for more information.

Once the migration finishes, the credit starts to be used. So I think it's time to remove them.

**Comment by PhysSong** [comment]
Since June 15th, 2021, travis-ci.org is now in read-only state. travis-ci.com builds are not enabled at this moment.
